staging/src/github.com/kubeedge/beehive/pkg/core/channel/context_channel.go

```diff
@@ -142,6 +134,14 @@ func (ctx *ChannelContext) SendSync(module string, message model.Message, timeou
 		ctx.anonChsLock.Unlock()
 	}()
 
+	sendTimer := time.NewTimer(timeout)
+	select {
+	case reqChannel <- message:
+	case <-sendTimer.C:
```

Comment:
Although this situation basically does not happen, but it is a great patch.

By the way, can you use `time.After` instead of `time.NewTimer` to streamline the code?

Reply:
I've made that change.

Reply:
now there's one `do-not-merge/invalid-commit-message` label, need to fix this

Reply:
kindly ping @ankitrgadiya , please squash the commits to one, and simplify the commit message to `beehive: fixes SendSync channel allocation behaviour`

Reply:
Got it, I'll make the change

Reply:
@zc2638 @gy95 Can you please re-review the changes?